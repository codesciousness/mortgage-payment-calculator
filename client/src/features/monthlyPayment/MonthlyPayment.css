.MonthlyPayment {
    display: flex;
    align-items: center;
    text-align: left;
}

@media only screen and (max-width: 1200px) {

    .MonthlyPayment {
        flex-direction: column;
    }
}

@media only screen and (max-width: 1000px) {

    .MonthlyPayment__title {
        text-align: center;
        margin: 1rem;
    }
}