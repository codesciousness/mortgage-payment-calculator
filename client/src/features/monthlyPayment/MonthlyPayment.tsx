import React from 'react';
import DonutChart from '../../components/donutChart/DonutChart';

type MonthlyPaymentProps = {
    
};

const MonthlyPayment = ({  }: MonthlyPaymentProps): JSX.Element => {

    return (
        <section id='MonthlyPayment'>
            <DonutChart/>
        </section>
    );
};

export default MonthlyPayment;