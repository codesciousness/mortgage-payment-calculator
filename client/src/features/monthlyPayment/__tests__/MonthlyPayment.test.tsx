import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import MonthlyPayment from '../MonthlyPayment';

jest.mock('../../donutChart/DonutChart', () => () => (<section data-testid="DonutChart">Donut Chart</section>));
jest.mock('../../donutChartLegend/DonutChartLegend', () => () => (<section data-testid="DonutChartLegend">Donut Chart Legend</section>));

describe('<MonthlyPayment />', () => {
    test('MonthlyPayment component is defined', () => {
        expect(<MonthlyPayment/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<MonthlyPayment/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('it renders correctly', async () => {
        render(<MonthlyPayment/>);
        expect(screen.getByText('Monthly payment breakdown')).toBeInTheDocument();
    });
    test('renders a DonutChart component', () => {
        render(<MonthlyPayment/>);
        expect(screen.getByTestId('DonutChart')).toBeInTheDocument();
        expect(screen.getByText('Donut Chart')).toBeInTheDocument();
    });
    test('renders a DonutChartLegend component', () => {
        render(<MonthlyPayment/>);
        expect(screen.getByTestId('DonutChartLegend')).toBeInTheDocument();
        expect(screen.getByText('Donut Chart Legend')).toBeInTheDocument();
    });
});