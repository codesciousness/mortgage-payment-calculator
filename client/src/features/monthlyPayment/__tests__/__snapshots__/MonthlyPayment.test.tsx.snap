// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthlyPayment /> component matches the snapshot 1`] = `
<section
  className="MonthlyPayment"
  id="MonthlyPayment"
>
  <div
    className="MonthlyPayment__donutChart"
  >
    <h2
      className="MonthlyPayment__title"
    >
      Monthly payment breakdown
    </h2>
    <section
      data-testid="DonutChart"
    >
      Donut Chart
    </section>
  </div>
  <section
    data-testid="DonutChartLegend"
  >
    Donut Chart Legend
  </section>
</section>
`;
